staging/src/k8s.io/component-base/logs/options.go

```diff
@@ -59,10 +59,14 @@ func (o *Options) Validate() []error {
 }
 
 // AddFlags add logging-format flag
-func (o *Options) AddFlags(fs *pflag.FlagSet) {
-	unsupportedFlags := fmt.Sprintf("--%s", strings.Join(unsupportedLoggingFlags(), ", --"))
-	formats := fmt.Sprintf(`"%s"`, strings.Join(logRegistry.List(), `", "`))
-	fs.StringVar(&o.LogFormat, logFormatFlagName, defaultLogFormat, fmt.Sprintf("Sets the log format. Permitted formats: %s.\nNon-default formats don't honor these flags: %s.\nNon-default choices are currently alpha and subject to change without warning.", formats, unsupportedFlags))
+func (o *Options) AddFlags(fs *pflag.FlagSet) error {
+	if unsupportedFlagsSet, unsupportedFlags := unsupportedLoggingFlags(); unsupportedFlagsSet != false {
```

Comment:
This should be checked in `Validate` and only if `logging-format` is non default